<!DOCTYPE html><html lang="en"><head><script async src="https://www.googletagmanager.com/gtag/js?id=UA-114897551-4"></script><script>window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'UA-114897551-4');
</script><script type="text/javascript" src="//platform-api.sharethis.com/js/sharethis.js#property=5ac2443d1fff98001395ab6c&amp;product=sticky-share-buttons" async="async"></script><title>Python Beginner Tutorial 8 - Exceptions | Coder Coacher - Coaching Coders</title><meta content="Python Beginner Tutorial 8 - Exceptions - All technical stuff in one place" name="description"><meta name="keywords" content="education, coding, programming, technology, nodejs, mongodb, software, computer science, engineering, teaching, coaching, coder, learning, java, kotlin"><meta name="viewport" content="width=device-width, initial-scale=1.0"><link rel="stylesheet" href="/css/font.css"><link rel="stylesheet" href="/css/bootstrap.css"><link rel="stylesheet" href="/css/style.css"><link rel="stylesheet" href="/css/coder-coacher.css"></head><body><div class="container-fluid"><h1 class="site-title"><a href="/">Coder Coacher</a></h1><hr><h4 class="site-subtitle text-right">Coaching Coders</h4></div><div id="amzn-assoc-ad-99d6751e-2392-4004-ad16-73aa8385d9d0"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=99d6751e-2392-4004-ad16-73aa8385d9d0"></script><div class="post__breadcrumb"><div class="container"><ol class="breadcrumb"><li><a href="/">Coder Coacher</a></li><li><a href="/DrapsTV/">DrapsTV</a></li><li class="active">â¤µ</li></ol></div></div><h2 class="post__title"><b>Python Beginner Tutorial 8 - Exceptions</b></h2><h5 class="post__date">2013-10-13</h5><div class="container"><div class="video-responsive"><iframe width="560" height="315" src="https://www.youtube.com/embed/-t0E5-xTbfs" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe></div><div class="post__text">welcome to Python beginner eight
exceptions every video will have old
slideshows and carried available in the
description exceptions is the concept
catching errors while the program is
running and doing something about it so
the user doesn't have to experience the
problem program crashing or not working
most often this happens with user
inputting incorrect data for our
variables or trying to open a file to
read that doesn't exist to avoid our
program crashing and getting no feedback
or not handling the issue we will learn
how to use the try statement and the
syntax is as follows try do stuff here
except do this stuff if any of the code
in the try has failed so we can surround
anything with this try and accept and we
only use it around things that can fail
so there's no point around putting it
around something that we've hard-coded
to set so let's create a little program
called - API that will try to open a
text file that doesn't exist cold-blood
or text when it can't open the file
instead of crashing we'll catch the
exception and print out that the file
could not be found
okay so let's come over to the NICS here
and let's create a new file call them to
8 pi and you know let's write our main
function
so def name brackets colon indent once
we write our try :
and then we indent twice and now let's
try and open a file called blah so your
f equals open open brackets blah dot txt
close quotes comma ah for read so we're
gonna try and open a file called blood
up tech txt I mean read mode it doesn't
exist
now for line in F so for each line in
the file that doesn't exist we want to
print line and we will strip it as well
strip so just good practice to our
backslash n so this is how we'd normally
write our file handling card and then we
close our file so f dot clothes for his
brackets and now we write our accept so
in the same in indentation level one in
from the left we write accept so that
the same indentation level is to try
however it's further down in the code we
do we are accept :
so this turd will execute if that
anything inside the try is failed so
let's just print out print the file was
either not found found or unable to be
read and then we close that quote and
now we've finished our program let's do
our if statement so if underscore
underscore name underscore underscore is
equal to open courts underscore
underscore main underscore closed quotes
: run main and our program is finished
let's save that so save that now let's
run it - 2 8 py and we get the file was
either not found or unable to be read
now
just like an if the try except block can
be put around anything so we put it
around all of our file code because
there is a chance that that file won't
exist so now if we wanted to create that
bla file so - yeah vim bla
dot txt and then we can start putting
some stuff in here so let's put in hi my
name name is traps and we'll save this
and now we've run our program again this
time we get our output and it doesn't
crash so but except never runs now one
optional word for our try/except is the
final each statement so this is optional
you don't have to use it when you use a
try and accept and it comes after the
exception block the finally will always
run the final block of code so the
syntax is try do this stuff except do
the stuff is to try a failed and finally
always do this stuff so let's let's
write a quick finally statement and add
on to our current current program so
open up our program and let's edit a
program this time let's try opening blah
one because we've already created a blur
file so let's make a blur one which
doesn't exist and then well after our
accept on the same indentation as the
accept we'll do a finally finally : so
this code will always run indent twice
you print exit in so this will always
run no matter what whether they try
fails or doesn't fail and let's save
this
and run our program so fire was either
not found or unable to be read so it's
exiting now we've opened up our program
again this time we'll edit it and put it
back to blah and we'll save and quit
save and quit and you'll notice that it
still prints out the exiting after it
prints out all of our lines in the file
so it's important to handle errors so
it's important to handle errors so the
end user doesn't have to see see them
crash your software so this is why we
use trials and accepts and we can also
use trials and accepts for more than
just outputting error messages but more
on that later this concludes our look at
exceptions in Python
don't fear if you don't remember all of
this you can easily come back and
re-watch this video comment consolidate
all the information if you have any
questions and you can't find the answer
after a quick Google search feel free to
leave them in the comments next we'll be
covering modules thanks for watching</div></div><div class="container-fluid bottom-ad"><div id="amzn-assoc-ad-6a809dda-347a-4187-8a86-91faf94575da"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=6a809dda-347a-4187-8a86-91faf94575da"></script></div><div class="text-center">We are a participant in the Amazon Services LLC Associates Program, an affiliate advertising program designed to provide a means for us to earn fees by linking to Amazon.com and affiliated sites.</div><script>(function(w, d){
    var b = d.getElementsByTagName('body')[0];
    var s = d.createElement("script"); s.async = true;
    var v = !("IntersectionObserver" in w) ? "8.6.0" : "10.4.2";
    s.src = "https://cdnjs.cloudflare.com/ajax/libs/vanilla-lazyload/" + v + "/lazyload.min.js";
    w.lazyLoadOptions = {};
    b.appendChild(s);
}(window, document));</script></body></html>